core-runner/src/runner.rs

```diff
@@ -189,6 +189,74 @@ pub struct Runner<MQ: MessageQueue, PS: ProgramStorage, WL: WaitList> {
     env: Environment<Ext>,
 }
 
+/// Builder for [`Runner`].
+pub struct RunnerBuilder<MQ: MessageQueue, PS: ProgramStorage, WL: WaitList> {
```

Comment:
Since this is needed just for tests, we ca use in-memory storage without generics